### PR TITLE
DFC-10908 - Composite FAC

### PR DESCRIPTION
**Pull request by snipered2003**

### CODE REVIEW

File: DFC.App.FindACourse/DFC.App.FindACourse.csproj [hunk 1]
```diff
@@ -12,15 +12,18 @@
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse/DFC.App.FindACourse.csproj]: Should we be targeting .Net core 3.1 as this is a new project

File: DFC.App.FindACourse.Repositories/FindACourseRepository.cs [hunk 2]
```diff
@@ -1,12 +1,35 @@
-﻿using System.Threading.Tasks;
+﻿using DFC.App.FindACourse.Data.Enums;
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse.Repositories/FindACourseRepository.cs]: Is this really a repository or is it a service?

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+﻿[*.cs]
```
Comment (Gurnaik-Randhawa) [inline .editorconfig]: There are around 260 warnings with this solution are we going to address these now or later?

File: DFC.App.FindACourse/Startup.cs [hunk 4]
```diff
@@ -32,7 +43,31 @@ public void ConfigureServices(IServiceCollection services)
...
+                {
+                    cfg.AddProfiles(
+                        new List<Profile>
+                        {
+                            new FindACourseProfile(),
+                        });
+                }).CreateMapper();
+            });
 
             services.AddSwaggerGen(c =>
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse/Startup.cs]: Do we need Swagger for this project?

File: DFC.App.FindACourse/ViewModels/SideBarViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+        public List<SelectListItem> DistanceOptions { get; set; }
+
+        public FiltersListViewModel CourseStudyTime { get; set; }
+
+        public FiltersListViewModel CourseHours { get; set; }
+        
+        public FiltersListViewModel CourseType { get; set; }
+        
+        public FiltersListViewModel StartDate { get; set; }
```
Comment (Gurnaik-Randhawa) [inline DFC.App.FindACourse/ViewModels/SideBarViewModel.cs]: Does the start date filters behave and hence use the same model as three above?

File: .editorconfig (hunk 3 above)
Comment (Gurnaik-Randhawa) [reply .editorconfig]: Also can't see the settings template file.

File: DFC.App.FindACourse/Startup.cs (hunk 4 above)
Comment (snipered2003) [reply DFC.App.FindACourse/Startup.cs]: This was there by default.  Removed as there is already one for the FAC API.

File: DFC.App.FindACourse/DFC.App.FindACourse.csproj (hunk 1 above)
Comment (snipered2003) [reply DFC.App.FindACourse/DFC.App.FindACourse.csproj]: Upgraded to 3.1. Bloody pain.

File: .editorconfig (hunk 3 above)
Comment (snipered2003) [reply .editorconfig]: Fixed most of the warnings.  Some have been ignored

Comment (snipered2003) [reply .editorconfig]: added settings file.

**Pull request merged**